index.js

```diff
@@ -123,63 +152,87 @@ module.exports = function (babel) {
-          'htmlbars-inline-precompile': 'default',
+          'htmlbars-inline-precompile': { export: 'default', shouldParseScope: false },
         };
 
         if (state.opts.modulePaths) {
           let modulePaths = state.opts.modulePaths;
 
-          modulePaths.forEach((path) => (modules[path] = 'default'));
+          modulePaths.forEach((path) => (modules[path] = { export: 'default' }));
         }
 
-        let modulePaths = Object.keys(modules);
-        let matchingModulePath = modulePaths.find((value) => t.isLiteral(node.source, { value }));
-        let modulePathExport = modules[matchingModulePath];
+        let presentModules = new Map();
 
-        if (matchingModulePath) {
-          let first = node.specifiers && node.specifiers[0];
-          let localName = first.local.name;
+        for (let module in modules) {
+          let paths = importDeclarations.filter(
+            (path) => !path.removed && path.get('source').get('value').node === module
+          );
 
-          if (modulePathExport === 'default') {
-            if (!t.isImportDefaultSpecifier(first)) {
-              let input = state.file.code;
-              let usedImportStatement = input.slice(node.start, node.end);
-              let msg = `Only \`import hbs from '${matchingModulePath}'\` is supported. You used: \`${usedImportStatement}\``;
-              throw path.buildCodeFrameError(msg);
+          for (let path of paths) {
+            let { node } = path;
+            let options = modules[module];
+
+            // Normalize 'moduleName': 'importSpecifier'
+            if (typeof options === 'string') {
+              options = { export: options };
             }
-          } else {
-            if (!t.isImportSpecifier(first) || modulePathExport !== first.imported.name) {
-              let input = state.file.code;
-              let usedImportStatement = input.slice(node.start, node.end);
-              let msg = `Only \`import { ${modulePathExport} } from '${matchingModulePath}'\` is supported. You used: \`${usedImportStatement}\``;
 
-              throw path.buildCodeFrameError(msg);
+            let modulePathExport = options.export;
+
+            let first = node.specifiers && node.specifiers[0];
+            let localName = first.local.name;
+
+            if (modulePathExport === 'default') {
+              let importDefaultSpecifier = node.specifiers.find((n) =>
+                t.isImportDefaultSpecifier(n)
+              );
+
+              if (!importDefaultSpecifier) {
+                let input = state.file.code;
+                let usedImportStatement = input.slice(node.start, node.end);
+                let msg = `Only \`import ${first.local.name} from '${module}'\` is supported. You used: \`${usedImportStatement}\``;
+                throw path.buildCodeFrameError(msg);
+              }
+            } else {
+              if (!t.isImportSpecifier(first) || modulePathExport !== first.imported.name) {
```

Comment:
This is going to bite us, we need to remove the order dependence.  It seems absolutely likely to me that `ember-template-imports` might want multiple imports, and if it does this code is going to error if the first one isn't the inline template compiler one?

Reply:
I was thinking about that, I'm happy to make this order independent, but it felt like a different PR. If you want I can roll it into this one though.

Reply:
Discussed offline, going to do this in a followup PR